 # Jupyter Lab Demo

**В Jupyter Lab можно устанавливать новые пакеты прямо в ячейках:**

In [1]:
%conda install numpy pandas seaborn scikit-learn

done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 22.11.1
  latest version: 24.7.1

Please update conda by running

    $ conda update -n base -c conda-forge conda

Or to minimize the number of packages updated during conda update use

     conda install conda=24.7.1



# All requested packages already installed.


Note: you may need to restart the kernel to use updated packages.


**Можно писать и выполнять код:**

In [21]:
import pandas as pd
import seaborn as sns
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

**Можно загружать и просматривать данные:**

In [11]:
data = pd.read_csv("Heart_Disease_Prediction.csv")
data.head()

index  Age  Sex  Chest pain type   BP  Cholesterol  FBS over 120  \
0      0   70    1                4  130          322             0   
1      1   67    0                3  115          564             0   
2      2   57    1                2  124          261             0   
3      3   64    1                4  128          263             0   
4      4   74    0                2  120          269             0   

   EKG results  Max HR  Exercise angina  ST depression  Slope of ST  \
0            2     109                0            2.4            2   
1            2     160                0            1.6            2   
2            0     141                0            0.3            1   
3            0     105                1            0.2            2   
4            2     121                1            0.2            1   

   Number of vessels fluro  Thallium Heart Disease  
0                        3         3      Presence  
1                        0         7       Absence  
2                        0         7      Presence  
3                        1         7       Absence  
4                        1         3       Absence

**Можно делать форматированные заметки:**

Датасет содержатся данные о 270 пациентах:

| Название столбца          | Описание                                                                 |
|---------------------------|--------------------------------------------------------------------------|
| Age                       | Возраст пациента.                                                        |
| Sex                       | Пол пациента.                                                            |
| Chest pain type           | Сила болей в груди, испытываемых пациентом                               |
| BP                        | Уровень артериального давления пациента.                                 |
| Cholesterol               | Уровень холестерина пациента.                                            |
| FBS over 120              | Результаты теста на уровень сахара в крови натощак выше 120 мг/дл.       |
| EKG results               | Результаты электрокардиограммы пациента.                                 |
| Max HR                    | Максимальная частота сердечных сокращений, достигнутая во время нагрузки.|
| Exercise angina           | Стенокардия, испытанная во время физической нагрузки.                    |
| ST depression             | Депрессия ST-сегмента на электрокардиограмме.                            |
| Slope of ST               | Наклон ST-сегмента на электрокардиограмме.                               |
| Number of vessels fluro   | Количество сосудов, видимых на флюороскопических снимках.                |
| Thallium                  | Результаты таллиевого стресс-теста.                                      |
| Heart Disease             | Диагностировано ли у пациента заболевание сердца.                        |


In [12]:
train_data, test_data = train_test_split(data)

In [18]:
model = LogisticRegression(max_iter=10000)
model.fit(train_data.drop("Heart Disease", axis=1), train_data["Heart Disease"])

LogisticRegression(max_iter=10000)

In [24]:
accuracy_score(model.predict(test_data.drop("Heart Disease", axis=1)), test_data["Heart Disease"])

0.8088235294117647